unlock-app/src/middlewares/lockMiddleware.js

```diff
@@ -3,123 +3,124 @@
 import React from 'react'
 import { LOCATION_CHANGE } from 'react-router-redux'
 import { CREATE_LOCK, SET_LOCK, WITHDRAW_FROM_LOCK, resetLock } from '../actions/lock'
-import { PURCHASE_KEY, addKey } from '../actions/key'
-import { SET_ACCOUNT, LOAD_ACCOUNT, CREATE_ACCOUNT, setAccount, resetAccountBalance } from '../actions/accounts'
+import { PURCHASE_KEY } from '../actions/key'
+import { SET_ACCOUNT, setAccount } from '../actions/accounts'
 import { setNetwork } from '../actions/network'
 import { setError } from '../actions/error'
 import { SET_PROVIDER } from '../actions/provider'
-import { REFRESH_TRANSACTION, addTransaction, refreshTransaction, updateTransaction, deleteTransaction } from '../actions/transaction'
+import { addTransaction, updateTransaction } from '../actions/transaction'
 
 import Web3Service from '../services/web3Service'
 
 // This middleware listen to redux events and invokes the services APIs.
+// It also listen to events from web3Service and dispatches corresponding actions
+// TODO: consider if on events we should only trigger more actions instead of calling web3Service directly
```

Comment:
I'd be strongly in favor of this. That way we can hook into these actions and change the logic over time. Doesn't need to be a now thing though.